### ISSUE TITLE
运行demo.py生成的result.mp4是一张图片播放数秒，不会动

### ISSUE THREAD

**Issue by Vijayue** [zhanglonghao1992/One-Shot_Free-View_Neural_Talking_Head_Synthesis#8]
大佬您好，感谢您的开源~

目前从voxceleb2里截取了几百个视频进行训练，然后存储模型

使用训练的模型运行demo.py，生成的MP4文件是一张图片播放了几秒钟，人物并不会动，请问您对这个问题有什么想法吗？

**Comment by zhanglonghao1992** [comment]
有没有先用crop-video.py处理一下driving video呢。最好把你的source, driving和result传上来我看下

**Comment by Vijayue** [comment]
刚才没有用crop_video.py处理，但处理后好像还是没有变化

source_image:

![0](https://user-images.githubusercontent.com/41794907/132468291-ac10a106-c190-4d5a-a368-f29a1cf5318c.png)

driving_video:

https://user-images.githubusercontent.com/41794907/132471510-096a5a9f-636e-47f0-bfbf-246ba5452919.mp4

result_video:

https://user-images.githubusercontent.com/41794907/132471770-6ca4c503-d505-4a35-b693-44d20a96fc7e.mp4

**Comment by Vijayue** [comment]
下面是driving_video和result视频的一些属性

<img width="555" alt="微信图片_20210908163930" src="https://user-images.githubusercontent.com/41794907/132476543-8947b095-e4e8-4fda-b001-7ad6af0cf07c.png">


predictions的输出，维度大小(53, 256, 256, 3)

<img width="249" alt="微信图片_20210908164027" src="https://user-images.githubusercontent.com/41794907/132476668-77381b5f-edb1-4e5f-8717-98532eff3112.png">

**Comment by zhanglonghao1992** [comment]
@Vijayue  This is what I got with your source image and driving video:

https://user-images.githubusercontent.com/17874285/132477992-2703d2db-503e-4e4d-9041-a33d111b2d25.mp4

**Comment by Vijayue** [comment]
Thank you for your test!

I found **keypoints_driving from different driving frames are the same (or close). ** Now I try to fix the error.

I guess maybe the model trained is not good.  It couldn't extract the correct keypoints from different frames.

Maybe the training dataset is too small, I just use about a hundred videos.

**Comment by panzhang0104** [comment]
I try the model using all voxceleb, It seems the output is the same as the source image not driving the image. 
![image](https://user-images.githubusercontent.com/75618833/133020359-327850fd-66d9-450b-ad6f-f6d01b570488.png)

**Comment by Vijayue** [comment]
@panzhang0104 I have settled the problem and found it's not the keypoints problem finally.
In the early epochs, the keypoints are not accurate. You maybe can try to train more time.

**Comment by panzhang0104** [comment]
Many Thanks!

**Comment by Vijayue** [comment]
@zhanglonghao1992 Excuse me, I experimented many times in different hyperparameters, but some losses are stable in all the training processes. The training is really hard. Would you please share your training log for my reference, thank you a lot.

**Comment by DWCTOD** [comment]
> @panzhang0104 I have settled the problem and found it's not the keypoints problem finally.
> In the early epochs, the keypoints are not accurate. You maybe can try to train more time.

您好，大佬训练了多久才可以。我这边用400个左右的视频训练了150个epoch，依然不动（已经训练了快一周了）。

**Comment by panzhang0104** [comment]
我不是train的video，我是抽frame，大概200W张，train了一个epoch

**Comment by vicdxxx** [comment]
> @Vijayue This is what I got with your source image and driving video:
> 
>  concat-18.mp4 
> set yaw = -30:
> 
>  yaw_-30.mp4

I use your beta model but cannot get good performance as you
Is model or config not match?
python demo.py --result_video data/result.mp4 --config config/vox-256.yaml --checkpoint checkpoint/15kp-ep119.pth.tar --source_image data/man.png --driving_video data/man.mp4 --relative --adapt_scale --find_best_frame

https://user-images.githubusercontent.com/14140837/140889457-99387295-36fc-4565-ab22-25c6a52d5a1a.mp4

**Comment by zhanglonghao1992** [comment]
@vicdxxx  Make sure that the following functions in ```demo.py``` and ```animate.py``` belong to the **beta version**：
1. get_rotation_matrix()
2. keypoint_transformation()
3. normalize_kp()

I just ```#``` the above functions in the new version.

**Comment by vicdxxx** [comment]
> @vicdxxx Make sure that the following functions in `demo.py` and `animate.py` belong to the **beta version**：
> 
> 1. get_rotation_matrix()
> 2. keypoint_transformation()
> 3. normalize_kp()
> 
> I just `#` the above functions in the new version.

I use beta verision of these functions, which did work. Thanks
Could I ask what is major diff of you new version with beta version functions?
and What do you think the potential of 256x256 resolution -> 512x512 or larger resolution?

**Comment by vicdxxx** [comment]
> > @vicdxxx Make sure that the following functions in `demo.py` and `animate.py` belong to the **beta version**：
> > 
> > 1. get_rotation_matrix()
> > 2. keypoint_transformation()
> > 3. normalize_kp()
> > 
> > I just `#` the above functions in the new version.
> 
> I use beta verision of these functions, which did work. Thanks Could I ask what is major diff of you new version with beta version functions? and What do you think the potential of 256x256 resolution -> 512x512 or larger resolution?

https://user-images.githubusercontent.com/14140837/140904658-476ad0bd-dc4a-44ba-8a61-1b0f35fe1d1d.mp4

**Comment by zhanglonghao1992** [comment]
@vicdxxx You can refer to the [update] in README to follow up the new version.

**Comment by tyrink** [comment]
> 

你好@Vijayue , 请问较早的epoch生成的可视化图像应该是怎样的呢？我前30个epoch的可视化图除了前四张人脸，其他都是全白或全黑，是不是有问题呢？而且我用训练完的模型运行demo.py，生成的MP4文件就是source_img播放了几秒，并没有被驱动，请问这种情况建议继续训练更多epoch吗？

**Comment by Vijayue** [comment]
@tyrink 不好意思，我已经忘记了较早应该是什么样子的，理论上是否应该和source image比较相似？

关于source没有被driving驱动，我现在觉得可能和训练时间有关系，也可能有别的原因……这个不太好判断。

作者人超级nice的在不断更新模型，用作者分享的模型呢🤔🤔

**Comment by huxian0402** [comment]
> 大佬您好，感谢您的开源~
> 
> 目前从voxceleb2里截取了几百个视频进行训练，然后存储模型
> 
> 使用训练的模型运行demo.py，生成的MP4文件是一张图片播放了几秒钟，人物并不会动，请问您对这个问题有什么想法吗？

@Vijayue 我最近尝试用这个代码在vox数据上训练，也遇到这个问题，“运行demo.py生成的result.mp4是一张图片播放数秒，不会动"，请教下，您当时是怎么解决的呢？
